src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/integration/PactTest.kt

```diff
@@ -5,75 +5,150 @@ import au.com.dius.pact.provider.junitsupport.Provider
+  private val deliusUser = AuthUser("8751622134", "delius", "BERNARD.BEAKS")
+
+  @BeforeEach
+  fun setup() {
+    // start with a clean slate for referrals. this can be removed once we
+    // no longer have dependency on the seed migrations (see setup method)
+    referralRepository.deleteAll()
+  }
 
   @TestTemplate
   @ExtendWith(PactVerificationSpringProvider::class)
   fun pactVerificationTestTemplate(context: PactVerificationContext) {
     context.verifyInteraction()
   }
 
-  @State("a draft referral can be created")
-  fun noop() {
-  }
+  @State("an intervention has been selected and a draft referral can be created")
+  fun `no referrals, using intervention id 98a42c61 from seed data`() {}
 
   @State("There is an existing draft referral with ID of ac386c25-52c8-41fa-9213-fcf42e24b0b5")
-  fun `use referral ac386c25 from the seed`() {
+  fun `create referral with the required id`() {
+    referralService.createDraftReferral(
+      user = deliusUser,
+      crn = "X123456",
+      interventionId = accommodationInterventionID,
+      UUID.fromString("ac386c25-52c8-41fa-9213-fcf42e24b0b5"),
+    )
   }
 
-  @State("a draft referral with ID dfb64747-f658-40e0-a827-87b4b0bdcfed exists")
-  fun `use referral dfb64747 from the seed`() {}
+  @State(
+    "a draft referral with ID dfb64747-f658-40e0-a827-87b4b0bdcfed exists",
+    "a single referral for user with ID 8751622134 exists"
+  )
+  fun `create a new draft referral for 'deliusUser' with a specific createdAt timestamp`() {
+    val referral = referralService.createDraftReferral(
+      user = deliusUser,
+      crn = "X862134",
+      interventionId = accommodationInterventionID,
+      overrideID = UUID.fromString("dfb64747-f658-40e0-a827-87b4b0bdcfed"),
+      overrideCreatedAt = OffsetDateTime.parse("2020-12-07T20:45:21.986389+00:00"),
+    )
+    referralRepository.save(referral)
+  }
 
   @State("a service category with ID 428ee70f-3001-4399-95a6-ad25eaaede16 exists")
-  fun `use service category 428ee70f from the seed`() {
-  }
+  fun `use service category 428ee70f from the real data migration`() {}
 
   @State("There is an existing draft referral with ID of d496e4a7-7cc1-44ea-ba67-c295084f1962, and it has had a service category selected")
-  fun `use referral d496e4a7 from the seed`() {}
+  fun `create a new draft referral with accommodation service category set (default)`() {
+    referralService.createDraftReferral(
+      user = deliusUser,
+      crn = "X123456",
+      interventionId = accommodationInterventionID,
+      UUID.fromString("d496e4a7-7cc1-44ea-ba67-c295084f1962"),
+    )
+  }
 
   @State("There is an existing draft referral with ID of d496e4a7-7cc1-44ea-ba67-c295084f1962, and it has had a service provider selected")
-  fun `use referral d496e4a7 from the seed, since it also has a service provider`() {}
+  fun `create a new draft referral with HARMONY_LIVING service provider (default)`() {
```

Comment:
This looks like it's doing the exact same thing as the function above. It looks like you have previously combined states to a single function(if i'm reading it correct). Could this also be done here? Even if so, not an issue.

Reply:
done